network/fuzz.go

```diff
@@ -0,0 +1,46 @@
+// +build gofuzz
+package network
+
+import (
+	"bytes"
+	"fmt"
+)
+
+func Fuzz(data []byte) int {
+
+	// deserialize
+	d1, err := Parse(data, ParseOpts{})
+	if err != nil {
+		return 0
+	}
+	if len(d1) == 0 && err != nil {
+		panic("d1 is empty but no err was returned")
```

Comment:
Please don't `panic`. Instead, pass the `*testing.T` to this function and call it's *Fatal()* method.